tests/components/admin_console/jobs/table.test.jsx

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {shallowWithIntl} from 'tests/helpers/intl-test-helper.jsx';
+import JobTable from 'components/admin_console/jobs/table.jsx';
+
+describe('components/admin_console/jobs/table', () => {
+    const createJobButtonText = (
+        <FormattedMessage
+            id='admin.complianceExport.createJob.title'
+            defaultMessage='Run Compliance Export Job Now'
+        />
+    );
+
+    const createJobHelpText = (
+        <FormattedMessage
+            id='admin.complianceExport.createJob.help'
+            defaultMessage='Initiates a Compliance Export job immediately.'
+        />
+    );
+
+    let baseProps;
+    const cancelJob = jest.fn(() => Promise.resolve({}));
+    const createJob = jest.fn(() => Promise.resolve({}));
+
+    beforeEach(() => {
```

Comment:
nit: why assign `baseProps` in `beforeEach` and not just as a `const` on line 25?

Reply:
Usually we assign empty functions on baseProp and mock specific functions in particular tests but that means mocking them multiple times for each test if needed. So instead i am mocking at base prop

The reason why i am doing with `beforeEach` is because we don't need to worry about test cases if we need to check for instances like `toHaveBeenCalledTimes` as they are assigned after every test.

I might have went a little overboard. Let me know if you see any downside with this.

Reply:
I could be mistaken, but I think the values for `cancelJob` and `createJob` persist across the tests, since they are only created once anyway -- i.e., you could assert `expect(createJob).toHaveBeenCalled();` in your last test and it would be true because it was called in the previous test.

My own preference is to make `baseProps` `const` and mock the callbacks of interest locally. IMO, it's easier to reason about mutables when they're "near by". It's perhaps a little verbose, but I think the pattern below works pretty well within each unit test:
```
const props = {
    ...baseProps,
    actions: {
        ...baseProps.actions,
        cancelJob: jest.fn(() => Promise.resolve({})),
    },
};
// ...
expect(props.actions.cancelJob).toHaveBeenCalled();
```

Alternatively, you could keep the `beforeEach` and just make sure to re-create the action callbacks each time. Maybe also inline `createJobButtonText` and `createJobHelpText`?

Reply:
> you could assert expect(createJob).toHaveBeenCalled(); in your last test and it would be true because it was called in the previous test.

It was failing if i asserted `expect(createJob).toHaveBeenCalled()` that in the last test so, i guess both our assumptions are wrong 

So, i checked by removing beforeEach and they still work perfectly fine. Just to be sure i copied tests couple of times and the assertions of `toHaveBeenCalledTimes(1)` works so i guess we don't even need to re-asssign them.

I removed before each and left the mocks at baseProps. I am 1/5 as a personal preference that mocks at base seem more readable as the test has fewer LOC. Let me know if you have stronger opinion on the readability of mocks in tests

Reply:
@lieut-data ☝️

Reply:
Aha, looks like `package.json` configures `"clearMocks": true` to ensure mocks are reset between tests. That's super useful to know in this context.

At this point, I think my feedback was just subjective, so I've got no more concerns, but I anticipate we may want to iterate on the question of top-level mock functions vs. local ones, as the behaviour here was surprising without the `clearMocks` context.

Reply:
> I anticipate we may want to iterate on the question of top-level mock functions vs. local ones

Sure, i would like to iterate on them too. I need to go back to docs or probably check how other popular projects use jest around to understand few things around mocking. I will get back on this with more info.